### ISSUE TITLE
PATH variable not set in PyCharm 2024.2.4

### ISSUE THREAD

**Issue by jstewmon** [fehnomenal/intellij-direnv#66]
👋 Hi, thank you for the excellent plugin. I'm having an issue with the latest version of PyCharm where my PATH variable is not reflecting the `PATH_add` instructions in my .envrc file. It looks like all other environment variables that are exported in my .envrc are populated, so I suspect there is some conflict specific to the PATH variable.

Build #PY-242.23726.102, built on October 22, 2024

**Comment by fehnomenal** [comment]
Hi,

the code has no special handling of PATH, so everything should work. Does the changes to PATH get picked up when using direnv in the terminal?

**Comment by jstewmon** [comment]
Yes, PATH is set as expected in the terminal. Sorry, I failed to mention that I'm observing a different PATH when with Run Configurations. Specifically, for python and pytest run configurations.

I am launching the IDE from the JetBrains toolbox. I can tell the IDE is loading my home shell profile (not from the project root) when it launches bc I have a command in my profile that requires a touchID.

Maybe there is some other IDE setting about loading the shell profile, or I need to launch the IDE differently?

**Comment by stijndcl** [comment]
> or I need to launch the IDE differently?

@jstewmon If you want your updated PATH to be available to your IDE you should launch it from the command-line after cd'ing into your desired directory (and letting direnv update your environment). E.g. cd into your project dir and type `pycharm`, `goland`, `webstorm`, ...

Agree that it would be nice for the plugin to support this instead.

**Comment by jstewmon** [comment]
@stijndcl thanks for your support on this issue. If I open my project by running `pycharm .`, it seems all of my direnv settings are inherited by the IDE, including updates to the path. But, having launched that way, when I try to use a run configuration with Enable Direnv and Trust .envrc checked, I get this error in the IDE:
```
Cannot invoke "java.util.Map.size()" because "m" is null
```

It would still be nice for the direnv plugin to work when the IDE is launched from a direnv directory, so that edits to environment variables can be picked up without having to restart the IDE.

**Comment by maxiem-ota-insight** [comment]
@fehnomenal 

I am also encountering this issue. It would seem that Pycharm, when configured to run with a virtual environment, takes it upon itself to overwrite all manual changes to the PATH variable, making it impossible to overwrite this by hand. I would also assume it takes away the ability of this plugin to do so.
